server/src/cache.ts

```diff
@@ -0,0 +1,61 @@
+import * as db from './lib/db';
+import { logInfo } from './lib/logger';
+
+const NodeCache = require('node-cache');
+
+export const dbCache = new NodeCache();
+
+export const getMapsCache = async () => {
+    const cachedMaps = dbCache.get('maps');
+
+    if (cachedMaps) {
+        return cachedMaps;
+    }
+
+    logInfo('mapsCache: Initializing cache');
+
+    const maps = await db.getMapsStats();
+    dbCache.set('maps', maps);
+    return maps;
+};
+
+export const addReplay = (metadata: any) => {
+    getMapsCache().then((cachedMaps: any) => {
+        const mapCacheMatch = cachedMaps.find((map: any) => map.mapUId === metadata.mapUId);
+
+        if (mapCacheMatch) {
+            logInfo('mapsCache: saveReplayMetadata: map found in cache, incrementing play count');
+            mapCacheMatch.count++;
+            mapCacheMatch.lastUpdate = metadata.date;
+        } else {
+            logInfo('mapsCache: saveReplayMetadata: map not found in cache, adding to cache');
+            cachedMaps.push({
+                mapUId: metadata.mapUId,
+                mapName: metadata.mapName,
+                count: 1,
+                lastUpdate: metadata.date,
+            });
+        }
+        dbCache.set('maps', cachedMaps);
+    });
+};
+
+export const deleteReplay = (replay?: any) => {
+    if (replay) {
+        getMapsCache().then((cachedMaps: any) => {
+            const mapCacheMatch = cachedMaps.find((map: any) => map.mapUId === replay.mapUId);
+            if (mapCacheMatch) {
+                mapCacheMatch.count--;
+                if (mapCacheMatch.count === 0) {
+                    logInfo('mapsCache: deleteReplayById: map play count is 0, deleting map from cache');
+                    cachedMaps.splice(cachedMaps.indexOf(mapCacheMatch), 1);
+                } else {
+                    logInfo('mapsCache: deleteReplayById: decrementing map play count');
+                }
+                dbCache.set('maps', cachedMaps);
+            }
+        });
+    } else {
+        logInfo('mapsCache: deleteReplayById: replay not found');
```

Comment:
This log should say that no replay was passed to the method if it's logged if `replay` is not defined. Is there even a reason why this replay parameter is optional? It seems kind of useless to call a deleteReplay method without a replay. Making it required would also remove the `if(replay)` and this log statement